src/main/java/org/jenkins/tools/test/PluginCompatTester.java

```diff
@@ -93,125 +87,100 @@ public void testPlugins() throws PluginCompatibilityTesterException {
                 new PluginCompatTesterHooks(config.getExternalHooksJars(), config.getExcludeHooks());
         // Determine the plugin data
 
-        // Scan bundled plugins. If there is any bundled plugin, only these plugins will be taken
-        // under the consideration for the PCT run.
-        UpdateSite.Data data = scanWAR(config.getWar(), "WEB-INF/(?:optional-)?plugins/([^/.]+)[.][hj]pi");
-        if (!data.plugins.isEmpty()) {
-            // Scan detached plugins to recover proper Group IDs for them. At the moment, we are
-            // considering that bomfile contains the info about the detached ones.
-            UpdateSite.Data detachedData = scanWAR(config.getWar(), "WEB-INF/(?:detached-)?plugins/([^/.]+)[.][hj]pi");
-
-            // Add detached if and only if no added as normal one
-            if (detachedData != null) {
-                detachedData.plugins.forEach((key, value) -> {
-                    if (!data.plugins.containsKey(key)) {
-                        data.plugins.put(key, value);
-                    }
-                });
-            }
-        }
-
-        if (data.plugins.isEmpty()) {
-            throw new PluginCompatibilityTesterException(
-                    "List of plugins to check is empty, it is not possible to run PCT");
+        String coreVersion = WarUtils.extractCoreVersionFromWar(config.getWar());
+        List<PluginMetadata> pluginMetadataList = WarUtils.extractPluginMetadataFromWar(
+                config.getWar(),
+                PluginMetadataHooks.loadExtractors(config.getExternalHooksJars()),
+                config.getIncludePlugins(),
+                config.getExcludePlugins());
+
+        // filter any plugins that are not being tested and group by Git URL
+        // and run through the pre-checkout hooks
+        // and group by Git URL
+        Map<String, List<PluginMetadata>> pluginsByrepo;
+        try {
+            pluginsByrepo = pluginMetadataList.stream()
+                    .map(new RunAndMapBeforeCheckoutHooks(pcth, coreVersion, config))
+                    .collect(Collectors.groupingBy(PluginMetadata::getGitUrl, HashMap::new, Collectors.toList()));
```

Comment:
Convention in this codebase is to prefer `LinkedHashMap` to `HashMap` to prefer the order specified by the user:

```suggestion
                    .collect(Collectors.groupingBy(PluginMetadata::getGitUrl, LinkedHashMap::new, Collectors.toList()));
```